### PR TITLE
Cleanup tests

### PR DESCRIPTION
**Pull request by mraspaud**
As a continuation of #84 , some more grunt work to refactor and cleanup the tests.

 - [ ] Closes #xxxx 
 - [ ] Tests added 
 - [ ] Tests passed 
 - [ ] Passes ``git diff origin/main **/*py | flake8 --diff`` 
 - [ ] Fully documented

### CODE REVIEW

**Comment by mraspaud** [comment]
Got the file under 2000 lines, :tada:

**Comment by mraspaud** [comment]
@djhoese I can't seem to figure out how to fix the sphinx error. I added the `pytest-lazy-fixture` to the `docs`, but the build doesn't seem to catch it. Could it be that the build image is being cached?

**Comment by djhoese** [comment]
Check how I do it in satpy. I think I had to load pytest_lazy_fixture directly and use tha5, not the pytest.plugin version.

**Comment by mraspaud** [comment]
@djhoese thanks! trying this now.

File: pycoast/tests/test_pycoast.py [hunk 1]
```diff
@@ -277,197 +270,205 @@ def cw_pil():
     return cw
 
 
-class TestContourWriterPIL(_ContourWriterTestBase):
-    """Test PIL-based contour writer."""
+@pytest.fixture(scope="session")
```
Comment (djhoese) [inline pycoast/tests/test_pycoast.py:273]: Are contour writers are immutable?

**Review by djhoese — APPROVED** [review]
If the coverage loss is purely just the changing lines in the test_pycoast.py module then :ok_hand:. If it is any part of the contour writer modules then not :ok_hand:.

Nice job. Looks much better.

**Comment by mraspaud** [comment]
I haven't changed anything in the production code, and didn't remove any test functionality, so my guess is that the coverage drop is due to fewer test lines and more production lines. Maybe we should remove the tests from the coverage computation?

**Comment by djhoese** [comment]
> Maybe we should remove the tests from the coverage computation?

I'm not a fan of this. I feel like it is a quick identifier for when a typo makes a test not run or a utility/helper in a test isn't used anymore.

**Comment by djhoese** [comment]
Coveralls shows that the `p_file_path` fixture is never used it seems.

**Comment by mraspaud** [comment]
> Coveralls shows that the `p_file_path` fixture is never used it seems.

Indeed, removing it.

**Pull request merged**